### ISSUE TITLE
ruby  on rails guide 5.1.2 kindle download link is broken

### ISSUE THREAD

**Issue by sagarkaurav** [rails/rails#29865]

**Comment by PHedkvist** [comment]
Just to give more details: The kindle download link found in the sidebar here http://guides.rubyonrails.org/ is broken, still for the version 5.1.4.

**Comment by khall** [comment]
Missing: http://guides.rubyonrails.org/v5.1/kindle/ruby_on_rails_guides_v5.1.4.mobi

Present: http://guides.rubyonrails.org/v5.0/kindle/ruby_on_rails_guides_v5.0.6.mobi
Present: http://edgeguides.rubyonrails.org/kindle/ruby_on_rails_guides_fced30c.mobi

**Comment by israelfaria** [comment]
The issue is still present. Any indicators of how to download it elsewhere?

**Comment by y-yagi** [comment]
I confirmed that `guides:generate:kindle` task occurred an error in 5-1-stable. 

```
$ gs
HEAD detached at upstream/5-1-stable
nothing to commit, working directory clean
$ bundle exec rake guides:generate:kindle
/home/yaginuma/.rbenv/versions/2.4.1/bin/ruby rails_guides.rb
Generating testing.md as testing.html
Generating action_cable_overview.md as action_cable_overview.html
Generating command_line.md as command_line.html
Generating _welcome.html.erb as _welcome.html
Generating asset_pipeline.md as asset_pipeline.html
Generating 3_2_release_notes.md as 3_2_release_notes.html
Generating caching_with_rails.md as caching_with_rails.html
Generating active_model_basics.md as active_model_basics.html
Generating active_support_core_extensions.md as active_support_core_extensions.html
Generating layouts_and_rendering.md as layouts_and_rendering.html
Generating action_view_overview.md as action_view_overview.html
Generating 5_1_release_notes.md as 5_1_release_notes.html
Generating active_job_basics.md as active_job_basics.html
Generating contributing_to_ruby_on_rails.md as contributing_to_ruby_on_rails.html
Generating getting_started.md as getting_started.html
Generating api_documentation_guidelines.md as api_documentation_guidelines.html
Generating 3_1_release_notes.md as 3_1_release_notes.html
Generating action_controller_overview.md as action_controller_overview.html
Generating active_record_migrations.md as active_record_migrations.html
Generating 4_2_release_notes.md as 4_2_release_notes.html
Generating active_support_instrumentation.md as active_support_instrumentation.html
Generating debugging_rails_applications.md as debugging_rails_applications.html
Generating rails_application_templates.md as rails_application_templates.html
Generating active_record_validations.md as active_record_validations.html
Generating i18n.md as i18n.html
Generating 4_0_release_notes.md as 4_0_release_notes.html
Generating active_record_postgresql.md as active_record_postgresql.html
Generating api_app.md as api_app.html
Generating engines.md as engines.html
Generating initialization.md as initialization.html
Generating association_basics.md as association_basics.html
Generating layout.html.erb as layout.html
Generating upgrading_ruby_on_rails.md as upgrading_ruby_on_rails.html
Generating active_record_querying.md as active_record_querying.html
Generating rails_on_rack.md as rails_on_rack.html
Generating 4_1_release_notes.md as 4_1_release_notes.html
Generating autoloading_and_reloading_constants.md as autoloading_and_reloading_constants.html
Generating 3_0_release_notes.md as 3_0_release_notes.html
Generating form_helpers.md as form_helpers.html
Generating active_record_basics.md as active_record_basics.html
Generating development_dependencies_install.md as development_dependencies_install.html
Generating 2_2_release_notes.md as 2_2_release_notes.html
Generating security.md as security.html
Generating active_record_callbacks.md as active_record_callbacks.html
Generating routing.md as routing.html
Generating action_mailer_basics.md as action_mailer_basics.html
Generating configuring.md as configuring.html
Generating 2_3_release_notes.md as 2_3_release_notes.html
Generating 5_0_release_notes.md as 5_0_release_notes.html
Generating generators.md as generators.html
Generating working_with_javascript_in_rails.md as working_with_javascript_in_rails.html
Generating plugins.md as plugins.html
Generating kindle/toc.html.erb as kindle/toc.html
=> Using output dir: /home/yaginuma/program/rails/master_y_yagi/rails/guides/output/kindle
=> Arranging html pages in document order
/home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:63:in `initialize': No such file or directory @ rb_sysopen - profiling.html (Errno::ENOENT)
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:63:in `open'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:63:in `block in generate_sections'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:61:in `each'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:61:in `each_with_index'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:61:in `generate_sections'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:23:in `block in generate'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:14:in `chdir'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/kindle.rb:14:in `generate'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/generator.rb:58:in `generate_mobi'
	from /home/yaginuma/program/rails/master_y_yagi/rails/guides/rails_guides/generator.rb:39:in `generate'
	from rails_guides.rb:27:in `<main>'
rake aborted!
Command failed with status (1): [/home/yaginuma/.rbenv/versions/2.4.1/bin/r...]
/home/yaginuma/program/rails/master_y_yagi/rails/guides/Rakefile:22:in `block (3 levels) in <top (required)>'
/home/yaginuma/program/rails/master_y_yagi/rails/guides/Rakefile:35:in `block (3 levels) in <top (required)>'
/home/yaginuma/.rbenv/versions/2.4.1/bin/bundle:23:in `load'
/home/yaginuma/.rbenv/versions/2.4.1/bin/bundle:23:in `<main>'
Tasks: TOP => guides:generate:html
(See full trace by running task with --trace)
```

I fixed this issue with 293be492be31ca78fbf414d64c8a397c92abbe35.  
Probably correctly generate kindle edition in next release of Rails.

**Comment by fxn** [comment]
The guides for v5.1.{2,3,4} in Kindle format have been generated manually.
